Automattic/jetpack
# Plugin scheduler: add update_plugins capability to /sites/%s endpoint

Fixes https://github.com/Automattic/dotcom-forge/issues/5943

## Proposed changes:

This adds a `update_plugins` capability to the default `/sites/%s` endpoint. Note that despite residing in Jetpack, this change is actually called on WPCOM as the [Jetpack end just returns false](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/sal/class.json-api-site-jetpack.php#L431). It's at this point on WPCOM end that decorate_response is called: fbhepr%2Skers%2Sjcpbz%2Schoyvp.ncv%2Serfg%2Spynff.jcpbz%2Qwfba%2Qncv.cuc%3Se%3Q5n863383%232835-og and the relevant decorate_response line [in Jetpack is this](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-site-endpoint.php#L967).

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?
NA

## Testing instructions:






* Apply this PR on the sandbox https://github.com/Automattic/jetpack/pull/36280#issuecomment-1985403981
* Test out both the `v1.1` and `v1.2` endpoints of `/sites/%s` for an Atomic site. `capabilities > update_plugins` should be `true`.
* Test out both the `v1.1` and `v1.2` endpoints of `/sites/%s` for a Simple site. `capabilities > update_plugins` should be `false`.

![CleanShot 2024-03-08 at 17 33 07](https://github.com/Automattic/jetpack/assets/533/c553e22a-b079-4993-866b-42e3677c7cc7)
